src/backend/Level0/CHIPBackendLevel0.hh

```diff
@@ -323,8 +352,30 @@ public:
 class CHIPContextLevel0 : public chipstar::Context {
   OpenCLFunctionInfoMap FuncInfos_;
   std::vector<LZEventPool *> EventPools_;
+  std::mutex CmdListMtx;
+  size_t NumCmdListsCreated_ = 0;
+  size_t CmdListsRequested_ = 0;
+  size_t CmdListsReused_ = 0;
+  std::stack<ze_command_list_handle_t> ZeCmdListRegStack_;
 
 public:
+  /**
+   * @brief Either return the immediate command list for this queue or a regular
+   * command list depending on if ICL is used or not. If ICL is not used, the
+   * regular command list can be created new or returnend from the stack of
+   * previously created handles.
+   *
+   * @return ze_command_list_handle_t
+   */
+  ze_command_list_handle_t getCmdListReg();
+
+  /**
+   * @brief reset the given command list and return it back to the stack
```

Comment:
```suggestion
   * @brief reset the given command list and return it back to the pool
```
Should also apply the change on the other occurrences of the "stack".

Reply:
see comment above